File: openapi/code_samples/javascript/api_v1_tools:search/post.js
import { ToolReadinessState } from '@scalekit-sdk/node'

const res = await scalekit.tools.searchTools('send a message to a slack channel', {
  identifier: 'user@example.com',
  topK: 10,
})

for (const tool of res.tools) {
  console.log(tool.name, tool.score)
  for (const connection of tool.connections) {
    // readinessState is a number at runtime -- always compare against the
    // named enum constant, never a raw number or a string.
    const isReady = connection.readinessState === ToolReadinessState.READY
    console.log(' ', connection.connectionName, isReady, connection.connectedAccountId)
  }
}

File: openapi/code_samples/python/api_v1_tools:search/post.py
from scalekit.v1.tools.tools_pb2 import TOOL_READINESS_STATE_READY

response = scalekit_client.tools.search_tools(
    query="send a message to a slack channel",
    identifier="user@example.com",
    top_k=10,
)

for tool in response[0].tools:
    print(tool.name, tool.score)
    for connection in tool.connections:
        # readiness_state is an int at runtime -- always compare against the
        # named enum constant, never a raw int or a string.
        is_ready = connection.readiness_state == TOOL_READINESS_STATE_READY
        print(" ", connection.connection_name, is_ready, connection.connected_account_id)
